### PR TITLE
Yamaha: Use annotations for OSGi DS. 

### PR DESCRIPTION
**Pull request by davidgraeff**
Signed-off-by: David Gräff <david.graeff@web.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaChannelsTypeProviderImpl.java [hunk 1]
```diff
@@ -19,12 +19,14 @@
...
 import org.eclipse.smarthome.core.thing.type.ChannelGroupTypeUID;
 import org.eclipse.smarthome.core.thing.type.ChannelType;
+import org.eclipse.smarthome.core.thing.type.ChannelTypeProvider;
 import org.eclipse.smarthome.core.thing.type.ChannelTypeUID;
 import org.eclipse.smarthome.core.thing.type.TypeResolver;
 import org.eclipse.smarthome.core.types.StateDescription;
 import org.eclipse.smarthome.core.types.StateOption;
 import org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants;
 import org.openhab.binding.yamahareceiver.handler.YamahaZoneThingHandler;
+import org.osgi.service.component.annotations.Component;
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaChannelsTypeProviderImpl.java]: The Travis build fails - you need to add an optional import to your manifest.

File: addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant.xml [hunk 2]
```diff
@@ -1,17 +1,7 @@
...
--->
-<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.yamahareceiver.discovery">
-   <implementation class="org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant"/>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant">
    <service>
       <provide interface="org.eclipse.smarthome.config.discovery.UpnpDiscoveryParticipant"/>
    </service>
-</scr:component>
+   <implementation class="org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant"/>
+</scr:component>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant.xml]: Missing newline

File: addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProviderImpl.xml [hunk 3]
```diff
@@ -1,18 +1,8 @@
...
--->
 <scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" name="org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProviderImpl">
-   <implementation class="org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProviderImpl"/>
    <service>
       <provide interface="org.eclipse.smarthome.core.thing.type.ChannelTypeProvider"/>
       <provide interface="org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProvider"/>
    </service>
-</scr:component>
+   <implementation class="org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProviderImpl"/>
+</scr:component>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProviderImpl.xml]: Missing newline

File: addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.internal.YamahaReceiverHandlerFactory.xml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<scr:component xmlns:scr="http://www.osgi.org/xmlns/scr/v1.1.0" immediate="true" name="org.openhab.binding.yamahareceiver.internal.YamahaReceiverHandlerFactory">
+   <service>
+      <provide interface="org.eclipse.smarthome.core.thing.binding.ThingHandlerFactory"/>
+   </service>
+   <reference bind="setChannelTypeProvider" interface="org.openhab.binding.yamahareceiver.internal.YamahaChannelsTypeProvider" name="ChannelTypeProvider" unbind="unsetChannelTypeProvider"/>
+   <implementation class="org.openhab.binding.yamahareceiver.internal.YamahaReceiverHandlerFactory"/>
+</scr:component>
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.internal.YamahaReceiverHandlerFactory.xml]: Missing newline

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaChannelsTypeProviderImpl.java [hunk 5]
```diff
@@ -45,41 +48,26 @@
             YamahaReceiverBindingConstants.BINDING_ID,
             YamahaReceiverBindingConstants.CHANNEL_PLAYBACK_PRESET_TYPE_NAMED);
 
-    /**
-     * {@inheritDoc}
```
Comment (martinvw) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaChannelsTypeProviderImpl.java]: Thanks for this cleanup :)

File: addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant.xml (hunk 2 above)
Comment (davidgraeff) [inline addons/binding/org.openhab.binding.yamahareceiver/OSGI-INF/org.openhab.binding.yamahareceiver.discovery.YamahaDiscoveryParticipant.xml]: Those files are auto generated, they don't have a license header as well ^^

**Comment by martinvw** [comment]
@kaikreuzer can you check these changes.

**Comment by kaikreuzer** [comment]
> For some reason, the framework creates a Thing instead of a Bridge for YamahaReceiverBindingConstants.BRIDGE_THING_TYPE. Is this just my ESH revision?

I just tested it:

screen shot 2017-08-08 at 11 23 55

Looks ok to me... I added the bridge through the Paper UI. Do you do it through a thing file? Did you maybe miss the "Bridge" keyword there?

File: addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaReceiverHandlerFactory.java [hunk 6]
```diff
@@ -59,7 +64,7 @@ protected ThingHandler createHandler(Thing thing) {
 
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
 
-        if (thingTypeUID.equals(YamahaReceiverBindingConstants.BRIDGE_THING_TYPE)) {
+        if (thingTypeUID.equals(YamahaReceiverBindingConstants.BRIDGE_THING_TYPE) && thing instanceof Bridge) {
```
Comment (kaikreuzer) [inline addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/YamahaReceiverHandlerFactory.java]: Before adding this check, we should figure out what is going on. I think we agree that such a check should not be necessary here (and if, that we would need some "else" branch to handle the faulty situation).

**Comment by davidgraeff** [comment]
I rebased ESH and oh and it worked. I'll remove the check

**Comment by davidgraeff** [comment]
I wanted to do this step by step this time, but because each step takes already very long, I do it in one go again. The all-in-one PR is #2533.

**Pull request closed without merging**